internal/compose/compose.go

```diff
@@ -275,20 +281,23 @@ func (p *Project) WaitForHealthy(opts CommandOptions) error {
 		return err
 	}
 
+	startTime := time.Now()
+	timeout := startTime.Add(waitForHealthyTimeout)
+
 	containerIDs := strings.Split(strings.TrimSpace(b.String()), "\n")
-	for {
+	for time.Now().Before(timeout) {
```

Comment:
Nit. Break the loop directly with a return so we don't have to remember to check for the timeout again after the loop? (And we are consistent on how we handle other end conditions as `SIGINT`).
```suggestion
	for {
		if time.Now().After(timeout) {
			return errors.New("timeout waiting for healthy container")
		}
```